tests/integration/test.changes.js

```diff
@@ -746,6 +746,86 @@ adapters.forEach(function (adapter) {
       });
     });
 
+    it('Immediately cancel changes', function () {
+      // fixes code coverage by ensuring the changes() listener
+      // emits 'complete' even if the db's task queue isn't
+      // ready yet
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        var changes = db.changes({live: true});
+        changes.on('error', reject);
+        changes.on('complete', resolve);
+        changes.cancel();
+      });
+    });
+
+    it('Changes with invalid ddoc view name', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: ''});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    it('Changes with invalid ddoc view name 2', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
+        var db = new PouchDB(dbs.name);
+        db.post({});
+        var changes = db.changes({live: true, filter: '_view', view: 'a/b/c'});
+        changes.on('error', resolve);
+        changes.on('change', reject);
+      });
+    });
+
+    if (adapter === 'local') {
+      // This test crashes CouchDB due to
+      // https://issues.apache.org/jira/browse/COUCHDB-2765
+      it('Changes with invalid ddoc with no map function', function () {
+        var db = new PouchDB(dbs.name);
+        return db.put({
+          _id: '_design/name',
+          views: {
+            name: {
+              empty: 'sad face'
+            }
+          }
+        }).then(function () {
+          return new PouchDB.utils.Promise(function (resolve, reject) {
+            var changes = db.changes({
+              live: true,
+              filter: '_view',
+              view: 'name/name'
+            });
+            changes.on('error', resolve);
+            changes.on('change', reject);
+          });
+        });
+      });
+    }
```

Comment:
Fair enough. If they don't plan on fixing it, then the fix on our side would be to put a `try/catch` around the `JSON.parse`, but I didn't want to do that since I thought they were going to be fixing it soon. I can add a TODO in a comment to let us know that we need to fix the bug on our side.

Also I'll note that this is a super rare bug that I doubt any users will actually run into, and either way if they do, they will either see an error from Couch or from Pouch.